src/depotchest.cpp

```diff
@@ -11,10 +11,10 @@ DepotChest::DepotChest(uint16_t type, bool paginated /*= true*/) :
     Container{type, items[type].maxItems, true, paginated}
 {}
 
-ReturnValue DepotChest::queryAdd(int32_t index, const Thing& thing, uint32_t count, uint32_t flags,
-                                 Creature* actor /* = nullptr*/) const
+ReturnValue DepotChest::queryAdd(int32_t index, std::shared_ptr<const Thing> thing, uint32_t count, uint32_t flags,
+                                 std::shared_ptr<Creature> actor /* = nullptr*/) const
 {
-	const Item* item = thing.getItem();
+	auto item = thing->getItem();
```

Comment:
Could this be `const auto`
```suggestion
	const auto item = thing->getItem();
```